journal/src/main/java/io/camunda/zeebe/journal/file/SegmentWriter.java

```diff
@@ -230,26 +229,24 @@ private void reset(final long index, final long lastFlushedIndex) {
     } catch (final BufferUnderflowException e) {
       // Reached end of the segment
     } catch (final CorruptedJournalException e) {
-      handleChecksumMismatch(e, nextIndex, lastFlushedIndex, position);
+      if (detectCorruptionAsPartialWrite) {
+        resetPartiallyWrittenEntry(e, position);
+      } else {
+        throw e;
+      }
     } finally {
       buffer.reset();
     }
   }
 
-  private void handleChecksumMismatch(
-      final CorruptedJournalException e,
-      final long nextIndex,
-      final long lastFlushedIndex,
-      final int position) {
-    // entry wasn't acked (likely a partial write): it's safe to delete it
-    if (nextIndex > lastFlushedIndex) {
-      FrameUtil.markAsIgnored(buffer, position);
-      buffer.position(position);
-      buffer.mark();
-      return;
-    }
-
-    throw e;
+  private void resetPartiallyWrittenEntry(final CorruptedJournalException e, final int position) {
+    LOG.debug(
+        "{} Found a corrupted or partially written entry at position {}. Considering it as a partially written entry and resetting the position.",
+        e.getMessage(),
+        position);
+    FrameUtil.markAsIgnored(buffer, position);
+    buffer.position(position);
```

Comment:
This is required, because the reading in the loop would have advanced the position into the corrupted entry. So we have to reset it to the beginning of this entry. Removing this fails the test `JournalTest::shouldDeletePartiallyWrittenEntry()`